app/controllers/bookmarks.js

```diff
@@ -26,5 +46,115 @@ export default Controller.extend({
+      [
+        'layer-groups',
+        'selectedZoning',
+        'selectedOverlays',
+        'selectedFirm',
+        'selectedPfirm',
+        'selectedCouncilDistricts',
+        'selectedLayerGroup',
+      ].forEach((selected) => {
+        queryParams[selected] = this.router.currentRoute.queryParams[selected]
+          ? JSON.parse(this.router.currentRoute.queryParams[selected])
+          : undefined;
+      });
+
+      const layerSet = {
+        id: crypto.randomUUID(),
+        name: 'New Saved Layer Set',
+        visibleLayers,
+        visibleLayerGroups,
+        queryParams,
+      };
+      this.set('savedLayerSets', [...this.savedLayerSets, layerSet]);
+      window.localStorage['saved-layer-sets'] = JSON.stringify(
+        this.savedLayerSets
+      );
+      this.track('bookmarkCurrentLayerSet');
+      // Hack to update the # which doesn't update automatically
+      document.querySelector('.badge.sup').innerText =
+        parseInt(document.querySelector('.badge.sup').innerText, 10) + 1;
+    },
+
+    deleteBookmarkedLayerSettings(id) {
+      this.set(
+        'savedLayerSets',
+        [...this.savedLayerSets].filter((lg) => lg.id !== id)
+      );
+      window.localStorage['saved-layer-sets'] = JSON.stringify(
+        this.savedLayerSets
+      );
+      this.track('deleteBookmarkedLayerSettings');
+      // Hack to update the # which doesn't update automatically
+      document.querySelector('.badge.sup').innerText =
+        parseInt(document.querySelector('.badge.sup').innerText, 10) - 1;
+    },
+
+    updateBookmarkedLayerSettings(id) {
+      const newLayerSets = [...this.savedLayerSets];
+      const updatedLayerSetIndex = newLayerSets.findIndex((lg) => lg.id === id);
+      newLayerSets[updatedLayerSetIndex].name =
+        document.getElementById('name').value;
+      this.set('savedLayerSets', newLayerSets);
+      window.localStorage['saved-layer-sets'] = JSON.stringify(
+        this.savedLayerSets
+      );
+      this.set('editMode', false);
+      // without the below, the name won't update in the dom
+      setTimeout(function () {
+        document.getElementById(id).innerText =
+          newLayerSets[updatedLayerSetIndex].name;
+      }, 1);
+      this.track('finishUpdateBookmarkedLayerSettings');
+    },
+
+    turnOnEditMode(id) {
+      this.set('editMode', id);
+      this.track('beginUpdateBookmarkedLayerSettings');
+    },
+
+    loadBookmarkedLayerSettings(bookmarkId) {
+      const layerToLoad = this.savedLayerSets.find(
+        (lg) => bookmarkId === lg.id
+      );
+      const layerGroups = [
+        ...this.router.currentRoute.parent.attributes.layerGroups,
+      ];
+      layerGroups.forEach((lg) => {
+        lg.visible = !!layerToLoad.visibleLayerGroups.includes(lg.id);
+        lg.layers.forEach((layer) => {
+          layer.visibility = !!layerToLoad.visibleLayers.includes(layer.id);
+        });
+      });
+
+      [
+        'selectedZoning',
```

Comment:
Can you move this list of strings that is repeated at line 63 into a variable? I see that line 64 adds `layerGroups`, so you could append that extra one there and use the variable for the rest.

Reply:
Done